client/jetpack-onboarding/main.jsx

```diff
@@ -142,7 +152,11 @@ class JetpackOnboardingMain extends React.PureComponent {
 }
 export default connect(
 	( state, { siteSlug } ) => {
-		const siteId = getUnconnectedSiteIdBySlug( state, siteSlug );
+		let siteId = getUnconnectedSiteIdBySlug( state, siteSlug );
+		if ( ! siteId ) {
+			siteId = getSiteId( state, siteSlug );
+		}
+
```

Comment:
We have a very strong dependency on some implicit conditions here, which we should at least document through a comment. The strongest one I think is that we're relying on something (probably the `Layout` component) to fetch data for all sites here (i.e. `<QuerySites allSites />`. Otherwise, we'd never be getting a `siteId` in the first place, which would make all subsequent logic fail.

Reply:
The most worrying part is that for a connected site, we can't even know whether a network request to get site data for the given `siteSlug` is ongoing, since before we have that site data, we don't know the `siteId`. so the `isRequestingSite()` check won't work 🙁

Reply:
Maybe we should fix `isRequestingSite()` to also `find( ..., siteIdOrSlug )`. Seems like a pretty fundamental flaw as it is now...